File: app/src/main/java/com/globalwarming/earthsaver/group/SearchAdapter.java
package com.globalwarming.earthsaver.group;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.globalwarming.earthsaver.R;
import com.globalwarming.earthsaver.User;

import java.util.ArrayList;
import java.util.List;

public class SearchAdapter extends RecyclerView.Adapter<SearchAdapter.ViewHolder> {

    private List<User> users = new ArrayList<>();
    private UserClickListener userClickListener;

    public void setList(List<User> users) {
        this.users.clear();
        this.users.addAll(users);
        notifyDataSetChanged();
    }

    public void setListener(UserClickListener userClickListener) {
        this.userClickListener = userClickListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_search_user, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        User user = users.get(position);

        holder.textViewName.setText(user.getName());
        holder.textViewEmail.setText(user.getEmail());
        holder.textViewAddress.setText(user.getLocation());

        holder.itemView.setOnClickListener(v -> {
            if (userClickListener != null) {
                userClickListener.onUserClick(user);
            }
        });
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        TextView textViewName;
        TextView textViewEmail;
        TextView textViewAddress;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            textViewName = itemView.findViewById(R.id.text_view_name);
            textViewEmail = itemView.findViewById(R.id.text_view_email);
            textViewAddress = itemView.findViewById(R.id.text_view_address);

        }

    }

}

interface UserClickListener {

    void onUserClick(User user);

}

File: app/src/main/java/com/globalwarming/earthsaver/group/SearchAdapter.kt
package com.globalwarming.earthsaver.group

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.globalwarming.earthsaver.User
import com.globalwarming.earthsaver.databinding.LayoutUserBinding
import java.util.*

class SearchAdapter : RecyclerView.Adapter<SearchAdapter.ViewHolder>() {

    private val users = ArrayList<User>()
    private var userClickListener: UserClickListener? = null

    fun setList(users: MutableList<User>) {
        this.users.clear()
        this.users.addAll(users)
        notifyDataSetChanged()
    }

    fun setListener(clickListener: UserClickListener?) {
        this.userClickListener = clickListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = LayoutUserBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val user = users[position]
        holder.binding.textViewName.text = user.name
        holder.binding.textViewPoints.text = "Points : ${user.points}"
        holder.binding.textViewAddress.text = user.location
        holder.itemView.setOnClickListener {
            userClickListener?.onUserClick(user)
        }
    }

    override fun getItemCount(): Int {
        return users.size
    }

    class ViewHolder(val binding: LayoutUserBinding) : RecyclerView.ViewHolder(binding.root)
}

interface UserClickListener {
    fun onUserClick(user: User)
}